public/app/features/dashboard/components/DashNav/iiris_maintenance_modal.ts

```diff
@@ -925,6 +925,19 @@ export class IirisMaintenanceModalCtrl {
           valid = false;
           this.scope.errorText += 'Toiston päättymisaika ei voi olla menneisyydessä. ';
         }
+        // Check if period continues over next DST change
+        const curYear = new Date().getFullYear();
+        const isCurrentlyDST = moment().isDST();
+        let nextChange;
+        if (isCurrentlyDST) {
+          nextChange = moment(curYear + '-10-01').endOf("month").startOf('isoWeek').subtract(1,'day').add(4,'hour');
+        } else {
+          nextChange = moment(curYear + '-03-01').endOf("month").startOf('isoWeek').subtract(1,'day').add(3,'hour');
+        }
```

Comment:
This is specific to EET timezone but I guess there are no suitable functions for this kind of checking in moment (or moment-timezone).

Reply:
Also this might change in future and might be easily overlooked because it is not tied to timezone data which would get updated when rules change.

Reply:
That's right, there may be problems with this, but I couldn't find ready function for checking next time change.